staging/src/k8s.io/component-helpers/scheduling/corev1/helpers.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package corev1
+
+import (
+	"encoding/json"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+// GetAvoidPodsFromNodeAnnotations scans the list of annotations and
+// returns the pods that needs to be avoided for this node from scheduling
+func GetAvoidPodsFromNodeAnnotations(annotations map[string]string) (v1.AvoidPods, error) {
+	var avoidPods v1.AvoidPods
+	if len(annotations) > 0 && annotations[v1.PreferAvoidPodsAnnotationKey] != "" {
```

Comment:
this is a "core/v1" helper that is deserializing an alpha annotation (`"scheduler.alpha.kubernetes.io/preferAvoidPods"`) on nodes? what is the graduation plan for this annotation and the data in it?

Reply:
That is a good question, but since it's being moved from [`pkg/apis/core/v1`](https://github.com/kubernetes/kubernetes/pull/92507/files#diff-f275cc5451de8d0a74dac8072e93d16cL500) I think that's consistent.

I don't know what the normal graduation plan for annotations is, does it differ from the rest of the api the annotation is a part of? I ask since this annotation is `alpha.kubernetes.io` but lives in `core/v1`, does that mean annotations may be versioned separately from their APIs?

Reply:
> since it's being moved from [`pkg/apis/core/v1`](https://github.com/kubernetes/kubernetes/pull/92507/files#diff-f275cc5451de8d0a74dac8072e93d16cL500) I think that's consistent

I'm looking at it from the perspective of external callers. At it's currently location, it shouldn't have any. Before we move it to a published repo intended to be consumed as a library, we should make sure the signature isn't one that will obviously need to be broken in order to keep working long-term, and this one clearly will (it's documentation states that it is serializing data to an annotation in anticipation of becoming a full-fledged field on the node status, which would require a signature change to continue extracting the data from the node properly)

Reply:
> Passing in a Node object and getting back v1.AvoidPods would seem more correct/stable since that could be updated to use a field rather than an alpha annotation.

+1 and a rename to `GetAvoidPodsFromNode()`

Reply:
> Passing in a Node object and getting back v1.AvoidPods would seem more correct/stable since that could be updated to use a field rather than an alpha annotation.

that would also mean the caller that specifically looks at the annotation value for validation purposes will likely need to continue doing so in some form (or this method would need to continue ensuring the annotation is valid if present)

Reply:
We are actually planing to deprecate the feature that uses this annotation, because what it achieves can be done with taints #94008

But this will take 3 or 4 releases to get removed.

Reply:
Update on this, that label is actually going to be deprecated apparently: https://github.com/kubernetes/kubernetes/issues/94008#issuecomment-704352146

Reply:
does that mean that the new publically exposed API function GetAvoidPodsFromNode* will become obsolete once the feature is removed?

Reply:
Correct. But we have other functions that we want to move to this staging repository. I don't think we are offering guarantees on the stability of this library. The guarantee is pretty similar to that of k8s.io/utils

Reply:
> I don't think we are offering guarantees on the stability of this library. The guarantee is pretty similar to that of k8s.io/utils

k8s.io/utils has presubmits in place to make sure the go API it exposes does not break (https://github.com/kubernetes/utils/blob/master/.github/workflows/test.yml)

Reply:
In that case, I suggest we pick a different function for this starting PR.

Reply:
if methods are moved to this new repository without stability guarantees, then possibly they may need to reside in specific sub packages e.g. "experimental" and readme files need to clarify that.

Reply:
>In that case, I suggest we pick a different function for this starting PR.


my thought as well, lets start with something else to unblock this PR, for example [GetPodPriority](https://github.com/kubernetes/kubernetes/blob/708dd2ea7a7765876cfb0a8329c90e92107860c5/pkg/api/v1/pod/util.go#L357)

Reply:
`GetAvoidPodsFromNodeAnnotations` replaced with `GetPodPriority`